src/content/1.7/development/components/grid-component.md

```diff
@@ -7,229 +7,404 @@ weight: 2
+        }
+
+        return $qb;
+    }
+    
+    // Get Count query builder that is used to get total count of all records (products)
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getBaseQuery();
+        $qb->select('COUNT(p.id_product)');
 
-## Grid services declaration
+        return $qb;
+    }
+    
+    // Base query can be used for both Search and Count query builders
+    private function getBaseQuery()
+    {
+        return $this->connection
+            ->createQueryBuilder()
+            ->from($this->dbPrefix.'product', 'p')
+            ->leftJoin(
+                'p',
+                $this->dbPrefix.'product_lang',
+                'pl',
+                'p.id_product = pl.id_product AND pl.id_lang = :context_lang_id AND pl.id_shop = :context_shop_id'
+            )
+            ->setParameter('context_lang_id', $this->contextLangId)
+            ->setParameter('context_shop_id', $this->contextShopId)
+        ;
+    }
+}
+```
 
-Only 3 services must be declared: the *Grid Factory*, the *Grid Definition Factory*, and the *Grid Data Factory*.
+Once Query builder is done, last step is to register it and configure `DoctrineGridDataFactory` to use it.
 
 ```yaml
-# In src/PrestaShopBundle/Resources/config/services/core/grid.yml
-
-# Grid Factory
-prestashop.core.grid.factory.foo:
-    class: 'PrestaShop\PrestaShop\Core\Grid\Factory\GridFactory'
+# Register ProductQueryBuilder
+prestashop.core.grid.query.product_query_builder:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Query\ProductQueryBuilder'
+    parent: 'prestashop.core.grid.abstract_query_builder'
     arguments:
-        - '@prestashop.core.grid.definition.factory.foo'
-        - '@prestashop.core.grid.data_factory.foo'
-        - '@form.factory'
-        - '@prestashop.core.hook.dispatcher'
-
-# Grid Definition Factory
-prestashop.core.grid.definition.factory.foo:
-    class: 'PrestaShop\PrestaShop\Core\Grid\Definition\Factory\FooGridDefinitionFactory'
-    parent: 'prestashop.core.grid.definition.factory.abstract_grid_definition'
+        - "@=service('prestashop.adapter.legacy.context').getContext().language.id"
+        - "@=service('prestashop.adapter.legacy.context').getContext().shop.id"
+    public: true
+    
+# Configure Grid Data factory to use query builder that we registered above
+prestashop.core.grid.data.factory.product_data_factory:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Data\Factory\DoctrineGridDataFactory'
+    arguments:
+        - '@prestashop.core.grid.query.product_query_builder' # service id of our query builder
+        - '@prestashop.core.hook.dispatcher' # every doctrine query builder needs hook dispatcher
+        - 'products' # this should match your grid id, in our case it's "products"
+```
+
+And that's it! Now we can use our Grid Data factory together with Search Criteria to get sorted, paginated and 
+filtered data for our Grid.
+
```

Comment:
also an example with DoctrineSearchCriteriaApplicatorInterface might be nice? wdyt?
or should it rather be in the "Lear more" section?

Reply:
i dont know really regarding applicator, as it is not actually "fully completed" yet. :) @mickaelandrieu wdyt?